### ISSUE TITLE
approle authentication does not take mount_point parameter

### ISSUE THREAD

**Issue by Akasurde** [ansible-collections/community.general#767]
_From @tcicone on Jun 17, 2019 20:40_

##### SUMMARY
approle authentication does not take the mount_point parameter, only ldap and user authentication uses it.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`lib/ansible/plugins/lookup/hashi_vault.py`

##### ANSIBLE VERSION
```
ansible 2.8.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Mar 26 2019, 22:13:06) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]


```

##### CONFIGURATION
Here's the broken code snippet:
```
    def auth_approle(self, **kwargs):
        role_id = kwargs.get('role_id', os.environ.get('VAULT_ROLE_ID', None))
        if role_id is None:
            raise AnsibleError("Authentication method app role requires a role_id")
 
        secret_id = kwargs.get('secret_id', os.environ.get('VAULT_SECRET_ID', None))
        if secret_id is None:
            raise AnsibleError("Authentication method app role requires a secret_id")
 
        self.client.auth_approle(role_id, secret_id)

```

##### OS / ENVIRONMENT
RHEL 7.6

##### STEPS TO REPRODUCE
Passing the mount_point parameter does nothing, uses the default auth/approle/login instead of our custom.

##### EXPECTED RESULTS
secrets are returned

##### ACTUAL RESULTS
Updating code to this works:
```
    def auth_approle(self, **kwargs):
        mount_point = kwargs.get('mount_point')
        role_id = kwargs.get('role_id', os.environ.get('VAULT_ROLE_ID', None))
        if role_id is None:
            raise AnsibleError("Authentication method app role requires a role_id")
 
        secret_id = kwargs.get('secret_id', os.environ.get('VAULT_SECRET_ID', None))
        if secret_id is None:
            raise AnsibleError("Authentication method app role requires a secret_id")
 
        self.client.auth_approle(role_id, secret_id, mount_point)
```

_Copied from original issue: ansible/ansible#57970_

**Comment by Akasurde** [comment]
_From @rlueckl on Aug 28, 2019 13:20_

I've had the same issue, so I made a PR. Should be pretty easy to merge. /cc @tcicone @samdoran

**Comment by Akasurde** [comment]
_From @gundalow on Sep 03, 2019 15:09_

resolved_by_pr: 61430

Review on https://github.com/ansible/ansible/pull/61430 welcome

**Comment by felixfontein** [comment]
!component =plugins/lookup/hashi_vault.py

**Comment by felixfontein** [comment]
CC @briantist

**Comment by briantist** [comment]
@Akasurde was the PR migrated? I don't see a new one linked here or in the other PR.

A lot has changed in the internals since that PR was opened anyway so we may need to reevaluate whether A) it's still needed and B) whether the fix is the same.

One of the things I noticed in the fix:
```python
role_id = kwargs.get('role_id', os.environ.get('VAULT_ROLE_ID', None))
```

We don't need to directly get environment variables anymore. We can just change the option spec to describe the env var(s) (and also ini var(s)) you want and the plugin loader will retrieve them correctly and with the correct precedence.

Happy to take a look if there's a new PR.
If not I could implement this fix myself probably, but it may be a few months as I'm not currently working directly on Vault in my job at the moment (but might be picking it up again soon).

**Comment by briantist** [comment]
I've reviewed the current code for this and I think this can be closed; it was actually fixed in #23 

https://github.com/ansible-collections/community.general/blob/main/plugins/lookup/hashi_vault.py#L415-L417

cc @felixfontein

**Comment by felixfontein** [comment]
resolved_by_pr #23
